c2corg_api/views/document_changes.py

```diff
@@ -0,0 +1,154 @@
+            Changes made by one user.
+
+            For more information about "continuation token pagination", see:
+            http://www.servicedenuages.fr/pagination-continuation-token (fr)
+        """
+        user_id = self.request.validated.get('u')
+        lang, token_id, token_time, limit = get_params(self.request)
+
+        if token_time is None:
+            token_time = datetime.datetime.now()
+
+        changes = get_changes_of_feed(token_id, token_time, user_id)
+
+        doc_ids = []
+        for ch in changes:
+            doc_ids.append(ch.id)
+
+        return load_feed(doc_ids)
+
+
+def get_changes_of_feed(token_id, token_time, user_id=None):
+    query = DBSession. \
+        query(HistoryMetaData.id, HistoryMetaData.user_id,
+              HistoryMetaData.written_at). \
+        order_by(HistoryMetaData.written_at.desc())
+
+    # pagination filter
+    if token_id is not None and token_time:
+        query = query.filter(
+            or_(
+                HistoryMetaData.written_at < token_time,
+                and_(
+                    HistoryMetaData.written_at == token_time,
+                    HistoryMetaData.id > token_id)))
+
+    if user_id is not None:
+        query = query.filter(HistoryMetaData.user_id == user_id)
+
+    return query.limit(limit).all()
+
+
+def load_feed(doc_ids):
+    doc_total = DBSession.query(DocumentVersion.id).count()
+
+    doc_changes = DBSession.query(DocumentVersion) \
+        .options(joinedload('history_metadata').
+                 joinedload('user').load_only(User.id, User.name,
+                                              User.username,
+                                              User.forum_username,
+                                              User.lang)) \
+        .options(joinedload('document')) \
+        .options(joinedload('document_locales_archive')) \
+        .order_by(desc(DocumentVersion.id)) \
+        .filter(DocumentVersion.history_metadata_id.in_(doc_ids)) \
+        .limit(limit) \
+        .all()
+
+    if not doc_changes:
+        return {'feed': []}
+
+    last_change = doc_changes[-1]
+    pagination_token = '{},{}'.format(
+        last_change.id,
+        urllib_parse.quote_plus(
+          last_change.history_metadata.written_at.isoformat()))
+
+    return {
+        'limit': limit,
+        'total': doc_total,
+        'count': doc_total,
+        'pagination_token': pagination_token,
+        'feed': [serialize_change(ch) for ch in doc_changes]
+    }
+
+
+def serialize_change(change):
+    return {
+        'written_at': change.history_metadata.written_at.isoformat(),
+        'lang': change.lang,
+        'document': {
+            'document_id': change.document.document_id,
+            'type': change.document.type,
+            'title': change.document_locales_archive.title,
+            'summary': change.document_locales_archive.summary,
+            'description': change.document_locales_archive.description
```

Comment:
I don't think we need the description (it can be very big, see for instance a route description) and probably not the summary either (especially if it has not changed from the previous version).

Here is the list of data requested by the client (see https://github.com/c2corg/v6_api/issues/605#issuecomment-275209675):
* title
* data
* author
* type of change (creation/update)
* comment about the change
Perhaps the quality would make sense as well.